### PR TITLE
Update user filter

### LINKED ISSUE #1424: User raitest has no plans listed under the Manage Plans option
- [ ] On Thailand preview environment, the user raitest is unable to see any plans while under the `Manage plans` option. The user however has plans available under the monitor tab.

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #1424 
Adds the following functionalities:
- For pages displaying OpenSrp plans push fetched user plans to `plans store` and also store user plan ids
- For super set plans, filter user plans from superset.

### CODE REVIEW

File: src/helpers/dataLoading/plans.ts [hunk 1]
```diff
@@ -41,7 +42,14 @@ export async function loadPlansByUserFilter<T>(
       }
       store.dispatch(actionCreator(response, userName));
       if (responseActionCreator) {
-        store.dispatch(responseActionCreator(response));
+        if (extractPlans) {
+          const extractedPlanRecords = response
+            .map(plan => extractPlanRecordResponseFromPlanPayload(plan as any))
```
Comment (moshthepitt) [inline src/helpers/dataLoading/plans.ts:47]: why `any`?

File: src/containers/pages/FocusInvestigation/active/index.tsx [hunk 2]
```diff
@@ -2,8 +2,12 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/active/index.tsx:2]: Let's add some tests to this component that codify the calls made to Superset when the user filter is turned on.

File: src/containers/pages/FocusInvestigation/active/tests/index.test.tsx [hunk 3]
```diff
@@ -502,7 +471,75 @@ describe('containers/pages/ActiveFocusInvestigation', () => {
     ).toEqual([selectedPlan1]);
   });
 
+  it('calls superset with the correct params', async () => {
+    const envModule = require('../../../../../configs/env');
+    envModule.DISPLAYED_PLAN_TYPES = 'FI,IRS,MDA,MDA-Point,Dynamic-FI,Dynamic-IRS,Dynamic-MDA'.split(
+      ','
+    );
+    envModule.ENABLE_DEFAULT_PLAN_USER_FILTER = false;
+    // export const DISPLAYED_PLAN_TYPES = ['FI', 'IRS', 'MDA', 'MDA-Point', 'Dynamic-FI'];
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/active/tests/index.test.tsx:480]: should we remove this?

File: src/containers/pages/FocusInvestigation/active/tests/index.test.tsx [hunk 4]
```diff
@@ -485,6 +433,27 @@ describe('containers/pages/ActiveFocusInvestigation', () => {
     await new Promise(resolve => setImmediate(resolve));
     wrapper.update();
 
+    expect(props.supersetService).toHaveBeenCalledTimes(1);
+    expect(props.supersetService).toHaveBeenCalledWith(0, {
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/active/tests/index.test.tsx:437]: Is it possible to get such an assertion that shows the userPlanIds explicitly in use?

Comment (ciremusyoka) [reply src/containers/pages/FocusInvestigation/active/tests/index.test.tsx:437]: yes

**Review by moshthepitt — APPROVED** [review]

**Review by p-netm — APPROVED** [review]

**Pull request merged**